source/common/config/metadata.cc

```diff
@@ -107,13 +107,12 @@ bool Metadata::metadataLabelMatch(const LabelSet& label_set,
 
 ConstMetadataSharedPoolSharedPtr
 Metadata::getConstMetadataSharedPool(Singleton::Manager& manager, Event::Dispatcher& dispatcher) {
-  return manager
-      .getTyped<SharedPool::ObjectSharedPool<const envoy::config::core::v3::Metadata, MessageUtil>>(
-          SINGLETON_MANAGER_REGISTERED_NAME(const_metadata_shared_pool), [&dispatcher] {
-            return std::make_shared<
-                SharedPool::ObjectSharedPool<const envoy::config::core::v3::Metadata, MessageUtil>>(
-                dispatcher);
-          });
+  return manager.getTyped<SharedPool::ObjectSharedPool<const envoy::config::core::v3::Metadata,
+                                                       MessageUtil, MessageUtil>>(
+      SINGLETON_MANAGER_REGISTERED_NAME(const_metadata_shared_pool), [&dispatcher] {
+        return std::make_shared<SharedPool::ObjectSharedPool<
+            const envoy::config::core::v3::Metadata, MessageUtil, MessageUtil>>(dispatcher);
+      });
```

Comment:
did this code actually changed? I think this is just reformatted. Revert?

Reply:
It's hard to read, but there's an extra template argument (see the repeated `MessageUtil`) for `EqualFunc`

Reply:
got it